### DISCUSSION TITLE
Welcome to HDF5-CSharp Discussions!

### DISCUSSION THREAD

**Discussion by LiorBanai** [LiorBanai/HDF5-CSharp#108] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by yoramfisher** [comment]
Hello - I am a noob. I downloaded the HDF5-CSharp project and built it in VS community 2022 (Version 17.1.2). It builds fine. It is targeting .NET Core 3.1 , Any CPU.    Many of the unit tests fail. 
For example: WriteAndReadListOfStrings() fails.   I have traced it down to to the low level code, and it seems writing strings to a dataset is not working.  The return value of the write command is 0.
Interestingly, WriteAndReadOneAsciiString() succeeds, but WriteAndReadOneString() fails.   I am running the original code unmodified. Was curious if there might be a good reason why some units tests fail right out of the gate.   Thanks.

**Reply by LiorBanai** [reply]
Hi @yoramfisher . You need to run it in x64 or 32 since it uses native calls so any cpu won't work.

**Reply by yoramfisher** [reply]
Wow. Thank-you for the fast reply.    I was not able to change the project "HDF5-CSharp.UnitTests" to platform x64.  It is an option in the dropdown list, but it immediately reverts back to Any CPU. And I have tried everything. Restart, unload project, rebuild etc.  So, I created a new project, set it successfully to x64, and added dependency to 'HDF5-CSharp' assembly. That Class Library IS built properly as x64.  The partial-text of my code is below:

     ```
 public void Go()
      {


         var testClass = new TestClassWithArray()
         {
            TestInteger = 2,
            TestDouble = 1.1,
            TestBoolean = true,
            TestString = "test string",
            TestDoubles = new double[] { 1.1, 1.2, -1.1, -1.2 },
            TestStrings = new string[] { "one", "two", "three", "four" }
         };
         long fileId = Hdf5.CreateFile("testFile.H5");

         Hdf5.WriteObject(fileId, testClass, "testObject");

         TestClassWithArray readObject = new TestClassWithArray();

         readObject = Hdf5.ReadObject(fileId, readObject, "testObject");

         Hdf5.CloseFile(fileId);

      }


The readObject has null for string (TestString) and null for TestStrings.  This seems to be confirmed with imageJ, plug in ilastik which also does not detect a string or string array in the file.

FWIW, I see that HDF5-CSharp uses HDF.PInvoke 1.10(1.10.500) and that HDF5-CSharp.UnitTests has a dependency on HDF.PIvoke 1.10(1.10.610).   

My project is built with Target Framework .NET 6.0, Target OS Windows, ... platform target x64.  C# appears to be 10.0. Builds with no serious warnings.

Again - thankyou for your help.
